concent_api/gatekeeper/views.py

```diff
@@ -31,14 +32,16 @@ def upload(request):
         return gatekeeper_access_denied_response('Unsupported content type.')
 
     path_to_file = request.get_full_path().partition(reverse('gatekeeper:upload'))[2]
-    response = parse_headers(request, path_to_file)
-    if response is not None:
-        logger.info(json.loads(response.content.decode())['message'])
-        return response
+    response_body = parse_headers(request, path_to_file)
+    if not isinstance(response_body, FileTransferToken.FileInfo):
+        logger.info(json.loads(response_body.content.decode())['message'])
+        return response_body
```

Comment:
In this `if` you should `assert` that `response_or_file_info` is indeed a response object - because if it is not (e.g. because someone updated`parse_headers()` to return something else in some case and forgot to update the view), we have a bug and we won't easily detect it.